### PR TITLE
Fix service/revision list output with '-o' param

### LINKED ISSUE #1273: List command doesn't return empty result in correct format
Hello, 
we are working on a knative plugin for IntelliJ and we found out this misbehavior.

### Bug report

`list` command doesn't return empty result in correct format (json, yaml, .).

```
$ kn service list -o json
No services found.
```

### Expected behavior

I would expect it in json format
E.g 
```
{
    "apiVersion": "serving.knative.dev/v1",
    "items": [],
    "kind": "Service",
    "metadata": {
        ...
    }
}
```

### Steps to reproduce the problem

Step above

### kn version

```
Version:      v0.21.0
Build Date:   2021-02-24 09:16:16
Git Revision: 5f68d612
Supported APIs:
* Serving
  - serving.knative.dev/v1 (knative-serving v0.21.0)
* Eventing
  - sources.knative.dev/v1alpha2 (knative-eventing v0.21.0)
  - eventing.knative.dev/v1beta1 (knative-eventing v0.21.0)
```

### Knative (serving/eventing) version

```
v0.21.0
```

### PR DESCRIPTION
**Pull request by evidolob**
## Description
This PR fixes behaviour of `service list -o <format>` and `revision list -o <format>` command in case when there are no services/revisions:
```
$ ./kn service list -o json
{
    "apiVersion": "serving.knative.dev/v1",
    "items": [],
    "kind": "ServiceList"
}
```

## Changes

* `revision/list.go` now checks `-o` option before type `No revisions found.` 
* `service/list.go` now checks `-o` option before type `No services found.` 

## Reference

Fixes #1273

### CODE REVIEW

**Comment by dsimansk** [comment]
Thanks!
/ok-to-test

File: pkg/kn/commands/revision/list.go [hunk 1]
```diff
@@ -81,8 +81,14 @@ func NewRevisionListCommand(p *commands.KnParams) *cobra.Command {
...
 
+			output, err := cmd.LocalFlags().GetString("output")
+			if err != nil {
+				fmt.Fprintf(cmd.OutOrStdout(), "output option not set properly\n")
+				return nil
+			}
+
 			// Stop if nothing found
-			if len(revisionList.Items) == 0 {
+			if output == "" && len(revisionList.Items) == 0 {
```
Comment (dsimansk) [inline pkg/kn/commands/revision/list.go:91]: The value can retrieved through `ListPrintFlag` that simplifies required changes a bit, e.g.:
```suggestion
			if !serviceListFlags.GenericPrintFlags.OutputFlagSpecified() && len(revisionList.Items) == 0 {
```

**Comment by maximilien** [comment]
@evidolob you might need to sign the CLA... see @google-cla bot message above

**Comment by evidolob** [comment]
@googlebot I signed it!

**Comment by evidolob** [comment]
@dsimansk I can review all `list` commands in this PR

**Comment by dsimansk** [comment]
> @dsimansk I can handle all `list` commands in this PR

If you don't mind the extra work, please go ahead. IMO it'd be better to align the listing output for every command.

**Comment by rhuss** [comment]
The tests are currently failing, so we can't include this PR for 0.22. However as this is a fix, we can quickly release a 0.22.1 afterwards when the e2e test issue is fixed.

**Comment by dsimansk** [comment]
@evidolob thanks!

I've gone through the PR and it covers all list commands. Except the newly added `domain` by me, I'll fix that in a follow-up PR.

/approve
/lgtm
/unhold

**Pull request merged**